tensorflow/python/keras/layers/preprocessing/preprocessing_stage_functional_test.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Functional preprocessing stage tests."""
+# pylint: disable=g-classes-have-attributes
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import time
+import numpy as np
+
+from tensorflow.python.data.ops import dataset_ops
+from tensorflow.python.keras import keras_parameterized
+from tensorflow.python.keras.engine import base_preprocessing_layer
+from tensorflow.python.keras.engine.input_layer import Input
+from tensorflow.python.keras.layers import convolutional
+from tensorflow.python.keras.layers import merge
+
+from tensorflow.python.keras.layers.preprocessing import image_preprocessing
+from tensorflow.python.keras.layers.preprocessing import normalization
+from tensorflow.python.keras.layers.preprocessing import preprocessing_stage_functional
+from tensorflow.python.keras.layers.preprocessing import preprocessing_test_utils
+from tensorflow.python.ops import array_ops
+from tensorflow.python.platform import test
+
+
+@keras_parameterized.run_all_keras_modes(always_skip_v1=True)
+class PreprocessingStageTest(
+    keras_parameterized.TestCase,
+    preprocessing_test_utils.PreprocessingLayerTest):
+
+  def test_adapt(self):
+
+    class PL(base_preprocessing_layer.PreprocessingLayer):
+
+      def __init__(self, **kwargs):
+        self.adapt_time = None
+        self.adapt_count = 0
+        super(PL, self).__init__(**kwargs)
+
+      def adapt(self, data, reset_state=True):
+        self.adapt_time = time.time()
+        self.adapt_count += 1
+
+      def call(self, inputs):
+        return inputs + 1
+
+
+    class PLMerge(PL):
+      def call(self, inputs):
+        return inputs[0] + inputs[1]
+
+    class PLSplit(PL):
+      def call(self, inputs):
+        return inputs + 1, inputs - 1
+
+    # Test with NumPy array
+    x0 = Input(shape=(3,))
+    x1 = Input(shape=(3,))
+    x2 = Input(shape=(3,))
+
+    l0 = PLMerge()
+    y = l0([x0, x1])
+
+    l1 = PLMerge()
+    y = l1([y, x2])
+
+    l2 = PLSplit()
+    z, y = l2(y)
+
+    stage = preprocessing_stage_functional.PreprocessingStageFunctional(
```

Comment:
Please add more types of models to increase test coverage:

- Functional PS that includes layer without an adapt method (e.g. Dense)
- Functional PS with a single input and single output
- Functional PS with a dict of inputs, e.g. `model = PS(inputs={'x1': x1, ...}, outputs=...)`